### PR TITLE
pods: make pod retry backoff accurate

### PR DESCRIPTION
**Pull request by dcbw**
Previously pods would set a retry deadline that backed off each
attempt. But the attempts were limited to every 30 seconds.

Change the retry loop to wake up when the next earliest pod's
deadline is, which helps process pod retries much closer to
their retry deadline than before.

### CODE REVIEW

File: go-controller/pkg/ovn/ovn.go [hunk 1]
```diff
@@ -478,15 +482,39 @@ func (oc *Controller) recordPodEvent(addErr error, pod *kapi.Pod) {
...
+	}
+	backoff := (re.backoffSec * time.Second) + (time.Duration(rand.Intn(500)) * time.Millisecond)
+	re.timeStamp = re.timeStamp.Add(backoff)
+}
+
 // iterateRetryPods checks if any outstanding pods have been waiting for 60 seconds of last known failure
 // then tries to re-add them if so
 // updateAll forces all pods to be attempted to be retried regardless of the 1 minute delay
-func (oc *Controller) iterateRetryPods(updateAll bool) {
+func (oc *Controller) iterateRetryPods(updateAll bool) time.Duration {
```
Comment (trozet) [inline go-controller/pkg/ovn/ovn.go:508]: update the doctext of what this is returning now. The next time we will wake up?

File: go-controller/pkg/ovn/ovn.go [hunk 2]
```diff
@@ -478,15 +482,39 @@ func (oc *Controller) recordPodEvent(addErr error, pod *kapi.Pod) {
 	}
 }
 
+const retryInitialDelay time.Duration = 2 * time.Second
+
+func newRetryEntry(pod *kapi.Pod, ignore bool) *retryEntry {
+	return &retryEntry{
+		pod:        pod,
+		timeStamp:  time.Now().Add(retryInitialDelay),
```
Comment (trozet) [inline go-controller/pkg/ovn/ovn.go:490]: it looks like time stamp was originally the time when the pod was added to the retry cache, but now it is the time when we are going to retry this pod. would be good to document this in the struct.

The retryPods map also has a comment that says: "// Map of pods that need to be retried, and the timestamp of when they last failed"

^this is no longer accurate

**Review by trozet — CHANGES_REQUESTED** [review]

File: go-controller/pkg/ovn/ovn.go (hunk 1 above)
Comment (dcbw) [reply go-controller/pkg/ovn/ovn.go:508]: done

File: go-controller/pkg/ovn/ovn.go (hunk 2 above)
Comment (dcbw) [reply go-controller/pkg/ovn/ovn.go:490]: done

**Pull request closed without merging**